src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,231 @@
 not provided by external packages and/or to satisfy particular needs of
 usage within ctapipe.
 """
-from astropy import units as u
+
+import logging
+from enum import Enum
+from pathlib import Path
+
+import astropy.units as u
 from astropy.coordinates import Angle, SkyCoord
+from astropy.table import Table
+from astropy.time import Time
+
+log = logging.getLogger("main")
 
 __all__ = ["get_bright_stars"]
 
 
-def get_bright_stars(pointing=None, radius=None, magnitude_cut=None):
+CACHE_FILE = Path("~/.psf_stars.ecsv").expanduser()
+
+
+class StarCatalogues(Enum):
+    Yale = "V/50/catalog"  # Yale bright star catalogue
+    Hippoarcos = "I/239/hip_main"  # hipparcos catalogue
+
+
+def select_stars(stars, pointing=None, radius=None, Bmag_cut=None, Vmag_cut=None):
+    """
+    utility function to cut stars based on magnitude and/or location
+
+    Parameters
+    ----------
+    stars: astropy table
+        Table of stars, including magnitude and coordinates
+    pointing: astropy Skycoord
+        pointing direction in the sky (if none is given, full sky is returned)
+    radius: astropy angular units
+        Radius of the sky region around pointing position. Default: full sky
+    Vmag_cut: float
+        Return only stars above a given apparent magnitude. Default: None (all entries)
+    Bmag_cut: float
+        Return only stars above a given absolute magnitude. Default: None (all entries)
+
+    Returns
+    -------
+    Astropy table:
+       List of all stars after cuts with same keys as the input table stars
     """
-    Get an astropy table of bright stars.
 
-    This function is using the Yale bright star catalog, available through ctapipe
-    data downloads.
+    if Bmag_cut is not None and "Bmag" in stars.keys():
```

Comment:
Are `Bmag` / `BTmag` and `Vmag` / `VTmag`  really the same?

Why was it needed to add the "T" versions?

Reply:
some catalogs only have the T versions available

Reply:
But we only deal with two here, which to my knowledge have Bmag and Vmag

Reply:
here is the link to the hippoarcos catalogue: https://vizier.cds.unistra.fr/viz-bin/VizieR-3?-source=I/239/hip_main&-out.max=50&-out.form=HTML%20Table&-out.add=_r&-out.add=_RAJ,_DEJ&-sort=_r&-oc.form=sexa

It has Vmag, but no Bmag. It has BTmag and VTmag available. The Yale catalogue is here:

https://vizier.cds.unistra.fr/viz-bin/VizieR-3?-source=V/50&-out.max=50&-out.form=HTML%20Table&-out.add=_r&-out.add=_RAJ,_DEJ&-sort=_r&-oc.form=sexa

Here it's only Vmag. 

the Nomad catalogue has both Bmag and Vmag:

https://vizier.cds.unistra.fr/viz-bin/VizieR-2

If it you want to not have the T versions i can remove it and add error messages when you try to use Bmag with the Yale and Hippoarcos catalogues.

Reply:
BT and VT are photometry measurements from Tycho catalog, the BT and VT bands are close to the Johnson's UBV, but not exactly the same. See par.2.2 here:
https://www.cosmos.esa.int/documents/532822/553204/sect2_02.pdf/9f2b5393-bc66-4480-89d2-88200aecf5c8

In case Bmag is not provided, but Vmag and B-V is there, you can calculate it. However, in our case, I believe this is not that much important. I suggest to let users decide on the selection and bear the responsibility of a correct use of the fields with an error raised in case the selection can't be calculated.

Reply:
Sounds good